pkg/zonemgr/zone_file.go

```diff
@@ -19,3 +23,26 @@ func NewZoneFile(fileName string) *ZoneFile {
 func (zoneFile *ZoneFile) writeFile(content string) (err error) {
 	return os.WriteFile(zoneFile.zoneFileFullName, []byte(content), zoneFilePerm)
 }
+
+func (zoneFile *ZoneFile) readFile() ([]byte, error) {
+	return os.ReadFile(zoneFile.zoneFileFullName)
+}
+
+func (zoneFile *ZoneFile) ReadSoaSerial() int {
+	serial := 0
```

Comment:
I would return `nil` in case the file doesn't exist. The logic to set 0 in cases there is no value in the existing file should reside in the ZoneCache.

Reply:
I can't return nil for the type int or string but I can return empty string if the file does not exist. Is that ok with you?

Reply:
done

Reply:
Imo it is not needed, we don't need to differentiate and have logic for both nil and 0 
just 0 is enough

Reply:
I don't agree. `ReadSoaSerial` should only read the value from the file, it shouldn't have any heuristic  of calculating the value in case the file doesn't exist, this logic should reside in the `ZoneCache` like the rest of the logic.

Reply:
you can name the function with something like CalculateSoaSerial
and then it will be responsible to give either 0 or nil, 
otherwise we are drifting with conversions and ptr that we can and should eliminate

i wont insist, but it is not correct imo, keep it simple is much better
maybe there is a middle way you can think of that will make it nicer and you will like ?